src/c2s/mongoose_c2s.erl

```diff
@@ -881,16 +923,14 @@ sm_unset_reason(_) ->
     error.
 
 %% @doc This is the termination point - from here stanza is sent to the user
--spec send_element(c2s_data(), [exml:element()] | exml:element(), mongoose_acc:t()) -> mongoose_acc:t().
-send_element(StateData = #c2s_data{host_type = <<>>}, El, Acc) ->
+-spec do_send_element(c2s_data(), exml:element(), mongoose_acc:t()) -> mongoose_acc:t().
+do_send_element(StateData = #c2s_data{host_type = <<>>}, El, Acc) ->
     send_xml(StateData, El),
     Acc;
-send_element(StateData = #c2s_data{host_type = HostType}, Els, Acc) when is_list(Els) ->
-    Res = send_xml(StateData, Els),
+do_send_element(StateData = #c2s_data{host_type = HostType}, #xmlel{} = El, Acc) ->
+    Res = send_xml(StateData, El),
```

Comment:
Yeah, about this, I remember we talked about it. But now I'm not that sure. The point was that calls to `exml:to_iolist/1` and the TCP socket send, are theoretically faster if batching, and that's why I wanted to optimise for them. But on the other hand, it seems like handlers like csi, amp, metrics, and stream-mgmt, might want to deal with the granularity of single elements, and it seems to complicate code too much to have sometimes to deal with lists and sometimes with single elements, of sometimes `#xmlel{}` and other times `mongoose_acc:t()`. So the change here was to have everything dealing with single elements of `mongoose_acc:t()`, and then the mongoose_c2s_acc helper transform this list of `mongoose_acc:t()` to a list of `gen_statem` actions, so that all code in c2s handles only one scenario.

So the old adage applies: first make it work, then make it beautiful, and only then if you really have to, make it fast. At the beginning I could force fast and pretty because I knew of few scenarios yet, but now with this thing becoming more complex... perhaps the minute optimisations can wait 🤷🏽